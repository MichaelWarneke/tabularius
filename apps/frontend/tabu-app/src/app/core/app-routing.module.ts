import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
