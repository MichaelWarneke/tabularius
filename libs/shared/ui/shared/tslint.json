{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
