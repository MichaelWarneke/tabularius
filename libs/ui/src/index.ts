export { TabuAppSidenavModule } from './lib/nav';
