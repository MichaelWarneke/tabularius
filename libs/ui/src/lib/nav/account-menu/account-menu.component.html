<button
  mat-icon-button
  [matMenuTriggerFor]="menu"
  aria-label="Toggle account menu"
>
  <mat-icon aria-label="Side nav toggle icon">account_circle</mat-icon>
</button>
<mat-menu #menu="matMenu">
  <button mat-menu-item routerLink="/account/login" *ngIf="!isAuth">
    Login
  </button>
  <button mat-menu-item routerLink="/account/logout" *ngIf="isAuth">
    Logout
  </button>
  <button mat-menu-item routerLink="/account" *ngIf="isAuth">Account</button>
</mat-menu>
