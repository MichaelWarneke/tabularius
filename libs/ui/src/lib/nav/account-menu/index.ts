export { AccountMenuModule } from './account-menu.module';
