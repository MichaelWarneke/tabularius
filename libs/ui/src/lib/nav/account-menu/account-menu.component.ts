import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'tabu-account-menu',
  templateUrl: './account-menu.component.html',
  styleUrls: ['./account-menu.component.scss']
})
export class AccountMenuComponent implements OnInit {
  @Input() isAuth = false;
  constructor() {}

  ngOnInit() {}
}
