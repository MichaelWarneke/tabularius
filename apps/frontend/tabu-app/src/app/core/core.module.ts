import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
