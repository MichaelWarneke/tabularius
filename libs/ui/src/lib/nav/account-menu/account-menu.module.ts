import { NgModule } from '@angular/core';
import { AccountMenuComponent } from './account-menu.component';
import { SharedModule } from '@tabularius/shared/ui';
import { RouterModule } from '@angular/router';

@NgModule({
  declarations: [AccountMenuComponent],
  imports: [SharedModule, RouterModule],
  exports: [AccountMenuComponent]
})
export class AccountMenuModule {}
