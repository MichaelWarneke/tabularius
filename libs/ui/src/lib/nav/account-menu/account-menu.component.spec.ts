import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AccountMenuComponent } from './account-menu.component';
import { SharedModule } from '@tabularius/shared/ui';

describe('AccountMenuComponent', () => {
  let component: AccountMenuComponent;
  let fixture: ComponentFixture<AccountMenuComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [AccountMenuComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AccountMenuComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
