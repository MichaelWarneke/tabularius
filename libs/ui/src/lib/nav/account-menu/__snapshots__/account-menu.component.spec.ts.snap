// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountMenuComponent should render 1`] = `
<tabu-account-menu
  isAuth="false"
>
  <button
    aria-haspopup="true"
    aria-label="Toggle account menu"
    class="mat-icon-button"
    mat-icon-button=""
    ng-reflect-menu="[object Object]"
  >
    <span
      class="mat-button-wrapper"
    >
      <mat-icon
        aria-hidden="true"
        aria-label="Side nav toggle icon"
        class="mat-icon notranslate material-icons mat-icon-no-color"
        role="img"
      >
        account_circle
      </mat-icon>
    </span>
    <div
      class="mat-button-ripple mat-ripple mat-button-ripple-round"
      matripple=""
      ng-reflect-centered="true"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button><mat-menu>
    
  </mat-menu>
</tabu-account-menu>
`;
